### PR TITLE
Changes from forum discussion

### PR DESCRIPTION
**Pull request by thrca**
The first commit fixes the issue when setting max_graphs to 0 (for unlimited) as the function was not checking for that before dumping with error.

The latter commits make the debug logging a bit more concise to figure out what is going on with the export process.  

I was able to get the tree view to reproduce the 0.8.8g behavior of creating a directory per tree branch, but this entailed a much more extensive rewrite.  Since this wasn't necessary for the mainline branch, I did not add that mess to the PR.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -724,11 +725,15 @@ function export_graphs(&$export, $export_path) {
...
 			}
 
+            export_debug('There are ' . sizeof($graphs) . ' graphs not in hosts to export for all trees');
+
 			$hosts = db_fetch_cell_prepared('SELECT GROUP_CONCAT(DISTINCT host_id)
 				FROM graph_tree_items
 				WHERE graph_tree_id IN(?)',
 				array(implode(', ', $ntree)));
 
+            export_debug('There are ' . sizeof(explode(',',$hosts)) . ' hosts to export for all trees');
```
Comment (netniV) [inline functions.php]: This code may incorrectly indicate that there is a host to be exported:

```php
php > echo sizeof(explode(',',false));
1
```

File: functions.php [hunk 2]
```diff
@@ -724,11 +725,15 @@ function export_graphs(&$export, $export_path) {
 				}
 			}
 
+            export_debug('There are ' . sizeof($graphs) . ' graphs not in hosts to export for all trees');
```
Comment (netniV) [inline functions.php]: Can you change the spaces to tabs to keep with the formatting standards?  There's even a line where you've not changed the code but seem to have introduced spaces before any text for some reason. 👍

File: functions.php [hunk 3]
```diff
@@ -741,6 +746,8 @@ function export_graphs(&$export, $export_path) {
 					}
 				}
 			}
+
+			export_debug('There are ' . sizeof($ngraph) . ' total graphs to export for all trees');
```
Comment (netniV) [inline functions.php]: I'm not sure if `$ngraph` is the correct variable name?

File: functions.php [hunk 4]
```diff
@@ -776,7 +783,7 @@ function export_graphs(&$export, $export_path) {
 
 			$exported++;
 
-			if ($exported >= $export['graph_max']) {
+			if ($exported >= $export['graph_max'] && $export['graph_max'] > 0) {
```
Comment (netniV) [inline functions.php]: This seems OK, though it could be shorted to `$export['graph_max'] && $exported >= $export['graph_max']`

**Comment by TheWitness** [comment]
@thrca, will you be making the updates that @netniV suggested?

File: functions.php (hunk 1 above)
Comment (thrca) [reply functions.php:735]: Yes, I was aware of this, but having some indicator was better than nothing at all for debugging.  I didn't find enough importance to attempt to solve this issue.

File: functions.php (hunk 2 above)
Comment (thrca) [reply functions.php:728]: vim settings.  Correcting.

File: functions.php (hunk 3 above)
Comment (thrca) [reply functions.php:750]: $ngraph array was added to if is_graph_allowed on line 743-744.  I think this is correct.

File: functions.php (hunk 4 above)
Comment (thrca) [reply functions.php:786]: Was intentionally done this way to be a more visible change to original code.  Preference?

**Comment by thrca** [comment]
There is still a finite limit to number of graphs before gexport fails, which did not happen in old 0.8.8 (pre-plugin) that I haven't figured out how to solve.. Basically, once we hit some specific number of graphs to export, I believe we run out of memory when adding to the array.  I have a heavily modified gexport that has removed a lot of extra stuff, but still suffers the issue, and I haven't had time to figure out how to solve it.

File: functions.php (hunk 2 above)
Comment (thrca) [reply functions.php]: Corrected.

**Comment by TheWitness** [comment]
We are making a change in 1.2.12 to make the memory limit unlimited for CLI scripts.  So, if you test on the 1.2.x branch, you should not have this issue.

File: functions.php (hunk 3 above)
Comment (TheWitness) [reply functions.php:750]: @thrca, can you change the "sizeof(" here to "cacti_sizeof("?  That will complete the requested changes.

Comment (TheWitness) [reply functions.php:750]: stet that, I still see soft-tabs.  We use a standard of hard tabs with a tab size of 4 chars.

```
set ts=4
```

File: functions.php (hunk 1 above)
Comment (netniV) [reply functions.php]: In a datacenter, that can lead to confusion where they think something should have appeared bu hasn't especially when they look at their rsync or ftp logs and find nothing happened.  It makes it appear as if something is broken, when in truth the display is simply wrong. Best to check the value is false or empty before trying to explode it.

File: functions.php (hunk 3 above)
Comment (netniV) [reply functions.php:750]: It looks like tabs are now being used but agree the sizeof should really be cacti_sizeof now.

**Comment by TheWitness** [comment]
This is all fixed up for the 2.0 release.

**Pull request closed without merging**